compiler/IREmitter/Payload/codegen-payload.c

```diff
@@ -1277,6 +1281,80 @@ VALUE sorbet_rb_array_empty(VALUE recv, ID fun, int argc, const VALUE *const res
     return Qfalse;
 }
 
+// This is the no-block version of rb_ary_uniq: https://github.com/ruby/ruby/blob/ruby_2_7/array.c#L5018-L5041
+SORBET_INLINE
+VALUE sorbet_rb_array_uniq(VALUE recv, ID fun, int argc, const VALUE *const restrict argv, BlockFFIType blk,
+                           VALUE closure) {
+    rb_check_arity(argc, 0, 0);
+    VALUE ary = recv;
+
+    VALUE hash, uniq;
+
+    if (RARRAY_LEN(ary) <= 1) {
+        hash = 0;
+        uniq = rb_ary_dup(ary);
+    }
+    else {
+        hash = sorbet_ary_make_hash(ary);
+        uniq = rb_hash_values(hash);
+    }
+    sorbet_rbasic_set_class(uniq, rb_obj_class(ary));
+    if (hash) {
+        sorbet_ary_recycle_hash(hash);
+    }
+
+    return uniq;
+}
```

Comment:
What do you think about moving this whole function into `array.c`? It doesn't look like it would benefit from being inlined, and as most of the work it does is in the vm already this would mean we'd cross the boundary only once per call.

Reply:
Makes sense! We don't want to do the same for `sorbet_rb_array_uniq_withBlock`, since we want opportunities to inline the block, right?

Reply:
Yep, exactly.